src/compiler/core.ts

```diff
@@ -1966,7 +1966,9 @@ namespace ts {
         const moduleKind = getEmitModuleKind(compilerOptions);
         return compilerOptions.allowSyntheticDefaultImports !== undefined
             ? compilerOptions.allowSyntheticDefaultImports
-            : moduleKind === ModuleKind.System;
+            : compilerOptions.ESModuleInterop

```

Comment:
should we make it an error to specify `--ESModuleInterop` with `--module` >= `ES2015`?

Reply:
When `module` >= `ES2015`, it's just a synonym for `allowSyntheticDefaultImports`. IMO it can be left as not-an-error, since you may still want it for interop if, eg, you're using babel for emit.

Reply:
but our output does not handle that case, and `--allowSyntheticDefaultImports` already does that..